Adds Wear specific section to README (#1366)
android/compose-samples: Jetcaster/README.md

- __Discover:__ [`com.example.jetcaster.ui.home.discover`](app/src/main/java/com/example/jetcaster/ui/home/discover)
- __Podcast Category:__ [`com.example.jetcaster.ui.category`](app/src/main/java/com/example/jetcaster/ui/home/category)

## Wear

> This sample showcases a 2 screens pager which allows to navigate between the Player and the Library.

Reviewer: ```suggestion
This sample showcases a 2-screen pager which allows navigation between the Player and the Library.
```

> From the library, users can access Latest episodes from followed podcasts, followed podcasts and queue.

Reviewer: ```suggestion
From the Library, users can access latest episodes from subscribed podcasts, and queue.
```

> From the podcast, users can access to episode details and add the episode to the queue.

Reviewer: ```suggestion
From the podcast, users can access episode details and add episodes to the queue.
```

From the Player screen, users can access a volume screen and a playback speed screen.

The sample implements [Wear UX best practices for media apps][mediappsbestpractices], such as:
- Support rotating side button (RSB) and Bezel for scrollable screens
- Display scrollbar on scrolling
- Display the time on top of the screens

The sample is built using the [Media Toolkit][[mediatoolkit]] which is an open source

> project part of [Horologist][horologist]  to ease the development of media apps on Wear OS built on top of Compose for Wear.

Reviewer: ```suggestion
project part of [Horologist][horologist] to ease the development of media apps on Wear OS built on top of Compose for Wear.
```

It provides ready to use UI screens, such the [EntityScreen][entityscreen]
that is used in this sample to implement many screens such as Podcast, LatestEpisodes and Queue. [Horologist][horologist] also provides
a VolumeScreen that can be reused by media apps to conveniently control volume either by interacting with the rotating side button(RSB)/Bezel or by
using the provided buttons.
For simplicity, this sample uses a mock Player which is reused across form factors,
if you want to see an advanced Media sample built on Compose that uses Exoplayer and plays media content,
refer to the [Media Toolkit sample][mediatoolkitsample].

The [official media app guidance for Wear OS][ [wearmediaguidance]]
advices to download content on the watch before listening to preserve power, this feature will be added to this sample in future iterations. You can
refer to the [Media Toolkit sample][mediatoolkitsample] to learn how to implement the media download feature.

### Architecture
The architecture of the Wear app is similar to the phone app architecture: each UI 'screen' has its 

> own [ViewModel][viewmodel] which exposes a `StateFlow<HomeViewState>` for the UI to observe.

Reviewer: ```suggestion
own [ViewModel][viewmodel] which exposes a `StateFlow<ScreenState>` for the UI to observe.
```
